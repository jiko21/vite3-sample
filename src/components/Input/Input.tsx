import { css } from '@emotion/react';
import React, { forwardRef, InputHTMLAttributes } from 'react';

type Props = InputHTMLAttributes<HTMLInputElement> & {
  testId?: string;
};

const style = css`
  box-sizing: border-box;
  border: solid 1px #cccccc;
  border-radius: 4px;
  line-height: 16px;
  padding: 8px 4px;
  height: 32px;
`;

export const Input = forwardRef<HTMLInputElement, Props>(({ testId, ...props }, ref) => (
  <input {...props} css={style} ref={ref} data-testid={`${testId}-input`} />
));
