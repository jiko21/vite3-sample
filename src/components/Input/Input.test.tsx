import { afterEach, describe, expect, it, vi } from 'vitest';
import { Input } from './Input';
import { cleanup, fireEvent, render, screen } from '@testing-library/react';

describe('components/Input', () => {
  afterEach(cleanup);

  it('correctly render', () => {
    const { container } = render(<Input />);
    expect(container).toMatchSnapshot();
  });

  it('correctly fire event', () => {
    const onChangeMock = vi.fn();
    render(<Input onChange={onChangeMock} testId="test" />);
    fireEvent.change(screen.getByTestId('test-input'), {
      target: {
        value: 'aaa',
      }
    });

    expect(onChangeMock).toBeCalled();
  });
});
