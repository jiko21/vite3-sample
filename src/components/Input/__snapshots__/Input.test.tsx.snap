// Vitest Snapshot v1

exports[`components/Input > correctly render 1`] = `
<div>
  <input
    class="css-68qs4z"
    data-testid="undefined-input"
  />
</div>
`;
